operations_on_datastructures/trie_multiple_search.cpp

```diff
@@ -0,0 +1,472 @@
+/**
+ * @file
+ * @brief [Trie datastructure](https://iq.opengenus.org/autocomplete-using-trie-data-structure/)
+ * with search variants
+ * @details
+ * This provides multiple variants of search functions
+ * on a trie structure utilizing STL. The trie is valid
+ * for only English alphabets.
+ * @author [Ghanashyam](https://github.com/g-s-k-zoro)
+ */
+
+#include <cassert>   /// for assert
+#include <cctype>    /// for tolower
+#include <cstring>   /// for string operations
+#include <iostream>  /// for IO Operations
+#include <queue>     /// for std::priority_queue
+#include <vector>    /// for std::vector
+/**
+ * @namespace operations_on_datastructures
+ * @brief Operations on data structures
+ */
+namespace operations_on_datastructures {
+/**
+ * @namespace trie_operations
+ * @brief Functions for Trie Multiple Search implementation.
+ */
+namespace trie_operations {
+
+/**
+ * @brief Class defining the structure of trie node and containing the methods
+ * to perform operations on them.
+ */
+class Tnode {
+ private:
+    static constexpr int ENGLISH_ALPHABET_SIZE = 26;

```

Comment:
Consider using `uint64_t`, `uint32_t`, `uint16_t` or even `uint8_t` for non-negative values. For negative values, use `int64_t` or their appropriate size. Check other parts of the code too ([reference](https://en.cppreference.com/w/cpp/types/integer)). 🙂

Reply:
Have used uint(s) where beneficial and corrected the documentation as suggested.